docs/api/design.md

```diff
@@ -0,0 +1,528 @@
+    no_noise_remove?: boolean; // Optional, default: false
+  }
+  ```
+- **Response**: `JobStatusResponse` model
+
+#### 4.2 Get Post-Processed Audio
+- **Endpoint**: `GET /api/episodes/{uuid}/postprocessed-audio`
+- **Description**: Retrieves the final post-processed audio
+- **Response**: Audio file stream
+
+### 5. Transcription Management
+
+#### 5.1 Get Annotated SRT
+- **Endpoint**: `GET /api/episodes/{uuid}/annotated-srt`
+- **Description**: Retrieves annotated SRT with speaker information
+- **Response**: Annotated SRT file
+
+#### 5.2 Update Annotated SRT
+- **Endpoint**: `PUT /api/episodes/{uuid}/annotated-srt`
+- **Description**: Updates annotated SRT
+- **Request Body**: `AnnotatedSrtUpdateRequest` model
+  ```typescript
+  {
+    content: string; // SRT content with annotations
+  }
+  ```
+- **Response**: `AnnotatedSrtUpdateResponse` model
+
+### 6. Metadata Generation
+
+#### 6.1 Generate Metadata
+- **Endpoint**: `POST /api/episodes/{uuid}/metadata`
+- **Description**: Generates chapter information and show notes from annotated SRT
+- **Request Body**: `MetadataGenerationRequest` model
+  ```typescript
+  {
+    audio_file_id?: number; // Optional specific audio file ID to use
+  }
+  ```
+- **Response**: `JobStatusResponse` model
+
+#### 6.2 Get Chapter Information
+- **Endpoint**: `GET /api/episodes/{uuid}/chapters`
+- **Description**: Retrieves chapter information in text format
+- **Response**: Chapter file content
+
+#### 6.3 Get Show Notes
+- **Endpoint**: `GET /api/episodes/{uuid}/show-notes`
+- **Description**: Retrieves generated show notes in markdown format
+- **Response**: Show notes content
+
+### 7. Export
+
+#### 7.1 Export Final Files
+- **Endpoint**: `POST /api/episodes/{uuid}/export`
+- **Description**: Generates final MP3, metadata, chapters, and SRT
+- **Request Body**: `ExportRequest` model
+  ```typescript
+  {
+    embed_chapters?: boolean; // Optional, default: true
+    include_show_notes?: boolean; // Optional, default: true
+    format?: string; // Optional: "mp3" | "wav", default: "mp3"
+  }
+  ```
+- **Response**: `JobStatusResponse` model
+
+#### 7.2 Download Exported Files
+- **Endpoint**: `GET /api/episodes/{uuid}/export/{file_type}`
+- **Description**: Downloads specific export file (mp3, metadata, chapters, srt)
+- **Path Parameters**: file_type: "mp3" | "chapters" | "show_notes" | "srt"
+- **Response**: File stream
+
+### 8. Audio Editing
+
+#### 8.1 Apply Audio Edits
+- **Endpoint**: `POST /api/episodes/{uuid}/audio-edits`
+- **Description**: Applies edits to audio files based on specified time ranges
+- **Request Body**: Skeleton for audio edits (implementation depends on web interface needs)
+- **Response**: `JobStatusResponse` model
+
+#### 8.2 Get Edited Audio Files
+- **Endpoint**: `GET /api/episodes/{uuid}/edited-audio`
+- **Description**: Retrieves all edited audio files
+- **Response**: List of `ProcessedFile` models
+
+### 9. Job Status Monitoring
+
+#### 9.1 Get Job Status
```

Comment:
I initially thought that the frontend could periodically fetch `/api/episodes` to check the `preprocess` and `postprocess` status. However, I realized that this is necessary for downloading files after `export`, so I decided to add it.

Reply:
Why not add the `exported` field to `Episode` just like `preprocessed` and `postprocessed`?

Reply:
I thought the `exported` field might be unnecessary since it can be determined by `metadata_generated`.

Reply:
then why do we still need job status?